code/builders/builder-manager/src/index.ts

```diff
@@ -80,8 +89,14 @@ export const getConfig: ManagerBuilder['getConfig'] = async (options) => {
     jsx: 'transform',
     jsxImportSource: 'react',
 
-    tsconfig: tsconfigPath,
-
+    tsconfigRaw: {
+      ...templateTsConfig,
+      ...(userTsConfig?.config ?? {}),
+      compilerOptions: {
+        ...(userTsConfig?.config.compilerOptions ?? {}),
+        ...templateTsConfig?.compilerOptions,
+      },
+    },
```

Comment:
@ndelangen This could be potentially pretty disruptive! WDYT? We could also just extract the `baseUrl` and the `paths`. On the other side, respecting the whole configuration of the user might be beneficial as well, because then esbuild gets more information about how to handle the Typescript code.